packages/sommelier.rb

```diff
@@ -2,90 +2,197 @@
+    
+    # Google's sommelier expects to find virtwl.h in their kernel source includes, but we may not have
+    # set of kernel headers which match, so we just download virtwl.h and then patch the sommelier source
+    # to look for the file locally.
+    ######################### Download virtwl.h from Chromium 5.4 kernel tree ###########################################
+    url_virtwl = "https://chromium.googlesource.com/chromiumos/third_party/kernel/+/5d641a7b7b64664230d2fd2aa1e74dd792b8b7bf/include/uapi/linux/virtwl.h?format=TEXT"
+    uri_virtwl = URI.parse url_virtwl
+    #filename_virtwl = File.basename(uri_virtwl.path)
+    filename_virtwl = 'virtwl.h_base64'
+    sha256sum_virtwl = 'a8215f4946ccf30cbd61fcf2ecc4edfe6d05bffeee0bacadd910455274955446'
+    
+    if File.exist?(filename_virtwl) && Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "Unpacking virtwl source code".yellow
+    else
+      puts "Downloading virtwl".yellow
+      system('curl', '-s', '-C', '-', '--insecure', '-L', '-#', url_virtwl, '-o', filename_virtwl)
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless
+        Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "virtwl archive downloaded".lightgreen
+      system 'mkdir -p build/linux'
+      system 'base64 --decode virtwl.h_base64 > build/linux/virtwl.h'
+    end    
+    
+    # lld is needed so libraries linked to system libraries (e.g. libgbm.so) can be linked against, since those are required for graphics acceleration.
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
+    
+    system "meson #{CREW_MESON_OPTIONS} -Dxwayland_path=#{CREW_PREFIX}/bin/Xwayland -Dxwayland_gl_driver_path=#{CREW_LIB_PREFIX}/dri -Ddefault_library=both -Dxwayland_shm_driver=noop -Dshm_driver=noop -Dvirtwl_device=/dev/null -Dpeer_cmd_prefix=\"#{CREW_PREFIX}#{PEER_CMD_PREFIX}\" build"
+    system "meson configure build"
+    system "ninja -C build"
+    
+    Dir.chdir ("build") do
+      system 'curl "https://chromium.googlesource.com/chromiumos/containers/sommelier/+/refs/heads/master/sommelierrc?format=TEXT" | base64 --decode > sommelierrc'
+      
+      # Following is to try to adhere to ChromeOS noexec
+      # set on the home dir as per 
+      # https://chromium.googlesource.com/chromiumos/docs/+/master/security/noexec_shell_scripts.md
+      system 'mkdir -p .sommelier'
+      system "echo 'wayland' >> .sommelier/CLUTTER_BACKEND"
+      system "echo ':0' >> .sommelier/DISPLAY"
+      system "echo 'x11' > .sommelier/GDK_BACKEND"
+      system "echo 'wayland-1' >> .sommelier/WAYLAND_DISPLAY"
+      system "echo '/var/run/chrome' >> .sommelier/XDG_RUNTIME_DIR"
+      # Changing this to 0.5 since otherwise everything looks small on a HiDPI screen.
+      system "echo '0.5' >> .sommelier/SCALE"
+      
+      system "touch .sommelier/MESA_LOADER_DRIVER_OVERRIDE"
+      case ARCH
```

Comment:
casing `ARCH` isn't ideal here, I'm sure theres a way to case the output of `uname -r` (which reports the kernel version)

Reply:
also, this is only done for `i686` and `x86_64`, when there are devices under 4.16 on arm

Reply:
also there are non-intel devices...

Reply:
The rationale for this is that mesa doesn't enable the i965 successor driver IRIS on kernels earlier than 4.16. So this is specific to intel machines. Do non-intel machines even use iris?

The intel discrete GPUS are only coming out in the next several months anyways, and the chances of those being used with arm machines are next to nil, no?

Reply:
Non-intel machines don't even use i965, and again, your just defaulting to disable iris when It's faster, you should be checking the kernel release with `uname -r`

Reply:
Suggestion for a test for kernel version > some version? Is there something builtin?

Reply:
@uberhacker, I don't recall there being a builtin, how should we go abt this?

Reply:
@ThatGeekyWeeb: I believe the way we sourced the variables in an earlier release was with a command alias to startsommelier.

Reply:
The "ruby" way to get the kernel version is `kernel_version = ``uname -r```.

Reply:
Perfect, @satmandu  simply do a `if` check with the above to see if the kernel is `4.16` or higher.

Reply:
I can do that, but can we confirm that MESA_LOADER_DRIVER_OVERRIDE is needed for arm kernels less than 4.16? Otherwise it is irrelevant.

(I can also revert the logic to load environment variables from multiple files.)

Reply:
I have no way to verify this, but I do know this may cause issues with non-Intel devices, if mesa tires to load the i965 driver on non-Intel device, it's likely going to just fail.

Reply:
I've modified the logic to check for kernel at env variable load time if on an x86 or x86_64 machine from ```~/.sommelier.env```, since a kernel update might make this irrelevant on such machines.

Reply:
Perfect!